src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -85,7 +107,100 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
         {
             AspireEventSource.Instance.DcpModelCreationStop();
         }
+    }
+
+    private async Task StartDashboardAsync(CancellationToken cancellationToken = default)
+    {
+        if (!distributedApplicationOptions.DashboardEnabled)
+        {
+            // The dashboard is disabled. Do nothing.
+            return;
+        }
+
+        if (_options.Value.DashboardPath is not { } dashboardPath)
+        {
+            throw new DistributedApplicationException("Dashboard path empty or file does not exist.");
+        }
+
+        var fullyQualifiedDashboardPath = Path.GetFullPath(dashboardPath);
+        var dashboardWorkingDirectory = Path.GetDirectoryName(fullyQualifiedDashboardPath);
+
+        var dashboardExecutableSpec = new ExecutableSpec
+        {
+            ExecutionType = ExecutionType.Process,
+            ExecutablePath = "dotnet",
+            Args = [fullyQualifiedDashboardPath],
+            WorkingDirectory = dashboardWorkingDirectory
+        };
+
+        var grpcEndpointUrl = await _dashboardServiceHost.GetResourceServiceUriAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
`GetResourceServiceUriAsync` could throw an error. Is error handling needed here? e.g.
```suggestion
        try
        {
            var grpcEndpointUrl = await _dashboardServiceHost.GetResourceServiceUriAsync(cancellationToken).ConfigureAwait(false);
        }
        catch (Exception ex)
        {
            throw new DistributedApplicationException("Error when getting resource service URL.", ex);
        }
```